sources/platform/storage/request_queue.md

```diff
@@ -274,6 +274,193 @@ Example payload:
+    requestHandler: async ({ enqueueLinks, request, $, log }) => {
+        log.info('enqueueing new URLs');
+        await enqueueLinks();
+
+        // Extract title from the page.
+        const title = $('title').text();
+        log.info(`New page with ${title}`, { url: request.loadedUrl });
+
+        // Save url of new URL and title to Dataset.
+        await Dataset.pushData({ url: request.loadedUrl, title });
+    },
+});
+
+await crawler.run(startUrls);
+
+await Actor.exit();
+```
+
+### Batch operations {#batch-operations}
+
+Request queues enhance performance and cut down on API calls and network latency for adding or removing individual requests by supporting batch operations.
+This efficiency booster enables you to enqueue or retrieve requests in bulk, enhancing efficiency when handling numerous URLs.
+Batch operations simplify process flows, decrease API calls, and reduce latency, thus accelerating your web crawling processes and enhancing reliability.
+
+<Tabs groupId="main">
+<TabItem value="JavaScript" label="JavaScript">
+
+```js
+const { ApifyClient } = require('apify-client');
+
+const client = new ApifyClient({
+    token: 'MY-APIFY-TOKEN',
+});
+
+const requestQueueClient = client.requestQueue('my-queue-id');
+
+// Add multiple requests to the queue
+await requestQueueClient.batchAddRequests([
+    { url: 'http://example.com/foo', uniqueKey: 'http://example.com/foo', method: 'GET' },
+    { url: 'http://example.com/bar', uniqueKey: 'http://example.com/bar', method: 'GET' },
+]);
+
+// Remove multiple requests from the queue
+await requestQueueClient.batchDeleteRequests([
+    { uniqueKey: 'http://example.com/foo' },
+    { uniqueKey: 'http://example.com/bar' },
+]);
+```
+
+</TabItem>
+<TabItem value="Python" label="Python">
+
+```python
+from apify_client import ApifyClient
+
+apify_client = ApifyClient('MY-APIFY-TOKEN')
+
+request_queue_client = apify_client.request_queue('my-queue-id')
+
+# Add multiple requests to the queue
+request_queue_client.batch_add_requests([
+    {'url': 'http://example.com/foo', 'uniqueKey': 'http://example.com/foo', 'method': 'GET'},
+    {'url': 'http://example.com/bar', 'uniqueKey': 'http://example.com/bar', 'method': 'GET'},
+])
+
+# Remove multiple requests from the queue
+request_queue_client.batch_delete_requests([
+    {'uniqueKey': 'http://example.com/foo'},
+    {'uniqueKey': 'http://example.com/bar'},
+])
+```
+
+</TabItem>
+</Tabs>
+
+### Distributivity {#distributivity}
+
+Our design facilitates the simultaneous processing of requests by multiple Actor runs or server instances, perfectly suiting large-scale web crawling projects.
+Distributivity ensures your scraping tasks can expand horizontally. The request queue includes locking mechanisms to avoid concurrent
+processing of the same request, a feature seamlessly integrated into Crawlee, requiring minimal extra setup. For more details, refer to the Crawlee documentation.
+
```

Comment:
I'd also mention that lock TTL should be set to the timeout used for request processing. Also, mention a page function timeout, or what it's called in Crawlee, as an example.